client/java/src/main/java/io/openlineage/client/Utils.java

```diff
@@ -0,0 +1,90 @@
+ * limitations under the License.
+ */
+
+package io.openlineage.client;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+import static org.apache.http.HttpHeaders.AUTHORIZATION;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import lombok.NonNull;
+import org.apache.http.client.methods.HttpRequestBase;
+
+/** Utilities class for {@link OpenLineageClient}. */
+public final class Utils {
+  private Utils() {}
+
+  private static final ObjectMapper MAPPER = newObjectMapper();
+
+  /** Returns a new {@link ObjectMapper} instance. */
+  public static ObjectMapper newObjectMapper() {
+    final ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(new Jdk8Module());
+    mapper.registerModule(new JavaTimeModule());
+    mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    mapper.disable(FAIL_ON_UNKNOWN_PROPERTIES);
+    mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+    return mapper;
+  }
+
+  /** Converts the provided {@code value} to a Json {@code string}. */
+  public static String toJson(@NonNull final Object value) {
+    try {
+      return MAPPER.writeValueAsString(value);
+    } catch (JsonProcessingException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  /** Converts the provided Json {@code string} to the specified {@code type}. */
+  public static <T> T fromJson(@NonNull final String json, @NonNull final TypeReference<T> type) {
+    try {
+      return MAPPER.readValue(json, type);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  /** Converts the provided {@code urlString} to an {@link URL} object. */
+  public static URL toUrl(@NonNull final String urlString) {
+    try {
+      final String urlStringWithNoTrailingSlash =
+          (urlString.endsWith("/") ? urlString.substring(0, urlString.length() - 1) : urlString);
+      return new URL(urlStringWithNoTrailingSlash);
+    } catch (MalformedURLException e) {
+      final AssertionError error = new AssertionError("Malformed URL: " + urlString);
+      error.initCause(e);
+      throw error;
+    }
+  }
+
+  /**
+   * Adds the provided {@code apiKey} as an {@code Authorization} HTTP header to the specified
+   * {@link HttpRequestBase} object.
+   */
+  public static void addAuthTo(
+      @NonNull final HttpRequestBase request, @NonNull final String apiKey) {
+    request.addHeader(AUTHORIZATION, "Bearer " + apiKey);
+  }
```

Comment:
I'd add a `TokenProvider` interface that the `HttpClient` can use that will allow users to supply tokens beyond just API keys. This would allow users to supply their own JWT provider that refreshed tokens on a periodic basis.

Reply:
How we get the API key is an implementation detail that doesn't really impact the structure of the code that sets the API key. e.g., in your example of the client builder, the underlying code doesn't really care where the `apiKey` comes from
```java
OpenLineageClient client = OpenLineageClient.builder()
  .url("http://localhost:5000")
  .apiKey("f38d2189-c603-4b46-bdea-e573a3b5a7d5")
  .build();
```
The key can come from an environment variable or a yaml config or whatever. 

Underneath, though, we can construct a `TokenProvider` implementation, such as a `StaticTokenProvider`, that simply returns the same string. Then at some point, we can support an `OauthTokenProvider` or something that generates a refreshable token so the value in the `Authorization` header can be whatever the `TokenProvider` provides. It's quick to implement as part of the original client implementation and allows us to support a concrete use case we know we'll need to support in the future.